### ISSUE TITLE
My group of issues

### ISSUE THREAD

**Issue by Fiszcz** [gidztech/docker-chromium#8]
Hi @gidztech !

**jest-puppeteer-docker** and **docker-chromium** are awesome solutions., but I have a few issues to this package. Sorry for that, I write them in one common issue.

Each of these single issues I will write as independent number:

## 1. **jest-puppeteer-docker** against security

My company has tried use your **jest-puppeteer-docker** package, but we have met some problems. We couldn't build **Dockerfile**, because there are some instructions which could not be executed properly in reason of our security rules (which we can meet in many financial institutions :( ). 

For example in my company we cannot download packages from github (in our case: **dumb-init** code), and we can fetch files like *.zip* / *.exe* only from our internal Nexus Repository service (so in our case we cannot download chromium from *https://commondatastorage.googleapis.com/*). 

I see two solutions for this case:

#### a.  Build and push (or find ready image) image to **Docker Hub** with all instructions from the current **Dockerfile** without only instructions with fetching of **chromium**. The second step would be possibility of customization for *download host address* for Chromium. 

Example exposed **Dockerfile** after changes:
```Dockerfile
FROM jest-puppeteer-docker:1.0

ENV REV=722234

ARG CHROMIUM_ADDITIONAL_ARGS
ENV CHROMIUM_ADDITIONAL_ARGS=${CHROMIUM_ADDITIONAL_ARGS}

RUN wget -q -O chrome.zip $CHROMIUM_DOWNLOAD_HOST/Linux_x64/$REV/chrome-linux.zip \
  && unzip chrome.zip \
  && rm chrome.zip \
  && ln -s $PWD/chrome-linux/chrome /usr/bin/google-chrome-unstable

RUN google-chrome-unstable --version

ENTRYPOINT ["/usr/bin/dumb-init", "--"]
CMD ["/usr/bin/entrypoint.sh"]
```

#### b. do not fetch **dumb-init** from github, but fetch as apt from Linux repository (``apt-get dumb-init``). And introduce **$CHROMIUM_DOWNLOAD_HOST** argument from **a.** variant solution

----

I think that **b.** variant solution would be easier. 

## 2. host.docker.internal

Good news!
In next version of Docker for Linux, ``host.docker.internal`` will work properly nad probably we will remove some parts of **entrypoint.sh**.

## 3 Custom download host for chromium

We should give possibility of changing for chromium download host.

We should introduce variable:
```javascript
const chromiumDownloadHost = process.env.PUPPETEER_DOWNLOAD_HOST || process.env.npm_config_puppeteer_download_host || process.env.npm_package_config_puppeteer_download_host || 'https://storage.googleapis.com/chromium-browser-snapshots';
```
and then pass it as arg by **docker-compose** to **Dockerfile**: ``--build-arg CHROMIUM_DOWNLOAD_HOST="${chromiumDownloadHost}"``

and then in **Dockerfile** immediately before ``RUN wget`` instruction:
```Dockerfile
ARG CHROMIUM_DOWNLOAD_HOST

RUN wget -q -O chrome.zip $CHROMIUM_DOWNLOAD_HOST/Linux_x64/$REV/chrome-linux.zip \
  && unzip chrome.zip \
  && rm chrome.zip \
  && ln -s $PWD/chrome-linux/chrome /usr/bin/google-chrome-unstable
```

## 4  ENV REV

We should refactor ``ENV REV`` in **Dockerfile**.

We should remove all this magic logic with replacing text in actually **Dockerfile** file:
```javascript
        const data = await readFile(dockerFilePath, { encoding: 'utf-8' });
        const previousTag = data.match(/REV=(.*)/)[1]; // get everything after revision on same line
        const newData = data.replace(previousTag, revision);
        await writeFile(dockerFilePath, newData, {
            encoding: 'utf-8'
        });
```
Instead of this we should pass argument to ``docker-compose build`` instruction as ``--build-arg CHROMIUM_REVISION=${revision}`` and then in **Dockerfile** we should add ``ARG CHROMIUM_REVISION`` immediately before ``RUN wget`` instruction:
```Dockerfile
ARG CHROMIUM_DOWNLOAD_HOST
ARG CHROMIUM_REVISION

RUN wget -q -O chrome.zip $CHROMIUM_DOWNLOAD_HOST/Linux_x64/$CHROMIUM_REVISION/chrome-linux.zip \
  && unzip chrome.zip \
  && rm chrome.zip \
  && ln -s $PWD/chrome-linux/chrome /usr/bin/google-chrome-unstable
```

## 5 https://commondatastorage.googleapis.com/

*https://commondatastorage.googleapis.com/* is old path to google resources, we should use *https://storage.googleapis.com/*. 

## 6 import-cert.sh

Could you explain me the purpose of **import-cert.sh** file ? Why do we need this ? 

## 7 entrypoint.sh

We should remove this part of **entrypoint.sh** file:
```bash
if [ -n "$CHROME_OPTS" ]; then
  CHROME_ARGS="${CHROME_ARGS} ${CHROME_OPTS}"
fi
```

because we do not create variable ``CHROME_OPTS`` anywhere. I know that this was copied from other solution for Docker **https://github.com/alpeware/chrome-headless-trunk**, but even there this variable is useless :D , because they copied code from other solution **https://github.com/nlnwa/docker-chrome-headless** and there variable ``CHROME_OPTS`` is used.

## 8 VOLUME

Do we need ``RUN mkdir /data`` and ``VOLUME /data`` instructions in **Dockerfile**  ?

----

----

That's all.

If you do not have anything against, I will work on these cases. :)

**Comment by gidztech** [comment]
Thanks for your interest in the project. 

I think the `import_cert.sh` file was copied from alpeware's config, but I don't know the purpose of it. We might be able to take it out. I believe `/data` is used by Chromium for the user directory when it's launched.

I will need spend a little longer going through your comment this weekend but overall it looks good to me and happy to accept contributions.

**Comment by Fiszcz** [comment]
Thanks for merges 🎉😀

You could also make release for these changes.

Now, I am working on usage **dumb-init** in some more easier way.

**Comment by gidztech** [comment]
> Thanks for merges 🎉😀
> 
> You could also make release for these changes.
> 
> Now, I am working on usage **dumb-init** in some more easier way.

Thanks for the PRs. You simplified a lot of stuff I was doing, e.g.  previously writing to the Dockerfile was silly. I was able to install `dumb-init` through aptitude, which I don't think was possible in the past, so I made a change there. I've now released a new version 1.3.0. Let's see if that helps with your issues.

**Comment by Fiszcz** [comment]
Awesome!

I am going to try to introduce visual tests to my company.

I close this issue.
